File: 1605/solution.py
class Solution:
    def restoreMatrix(self, rowSum: list[int], colSum: list[int]) -> list[list[int]]:
        ROWS, COLS = len(rowSum), len(colSum)
        res = [[0 for _ in range(COLS)] for _ in range(ROWS)]
        for row in range(ROWS):
            for col in range(COLS):
                res[row][col] = min(rowSum[row], colSum[col])
                rowSum[row] -= res[row][col]
                colSum[col] -= res[row][col]
        return res

File: 1605/solution.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int ROWS = rowSum.length;
        int COLS = colSum.length;
        int[][] res = new int[ROWS][COLS];

        for (int row = 0; row < ROWS; row++) {
            for (int col = 0; col < COLS; col++) {
                res[row][col] += Math.min(rowSum[row], colSum[col]);
                rowSum[row] -= res[row][col];
                colSum[col] -= res[row][col];
            }
        }
        return res;

    }
}

// time O(mn)
// space O(1)